### ISSUE TITLE
Validation schema accepts invalid values when `instanceof` is used

### ISSUE THREAD

**Issue by SpaceK33z** [webpack/webpack#3231]
**I'm submitting a bug report**

**Webpack version:**
2.1.0-beta.25

**Current behavior:**

The validation schema does not work for options that are using `instanceof`. As an example, I will show you the `modules.exprContextRegExp` option.

In the validation schema, this is defined for `modules.exprContextRegExp`:

```json
"exprContextRegExp": {
  "instanceof": "RegExp"
},
```

Now take this partial webpack config:

```js
modules: {
  exprContextRegExp: 1337
}
```

When running webpack, this passes the validation. A number clearly isn't a `RegExp` however. Even if passed a string, function or whatever, it will always pass validation.

This does not apply only for this specific option, but for **all options that use `instanceof`**. Whatever is used as option, validation will always pass.

**Expected/desired behavior:**

I would expect the option above (and other options that are defined in the schema with `instanceof`) to not pass validation.

**Comment by SpaceK33z** [comment]
@gajus, do you have any idea about this?

**Comment by SpaceK33z** [comment]
It seems like `instanceof` is ignored because you need to use the [ajv-keywords](https://github.com/epoberezkin/ajv-keywords) package for that to work.

**Comment by SpaceK33z** [comment]
Fixed this issue in #3232. @gajus, I would appreciate it if you could verify that this is the correct way to go.

**Comment by gajus** [comment]
Looks right to me. Just in case, CC @epoberezkin.
